### ISSUE TITLE
Slug Translation breaking pagination on custom post type archives

### ISSUE THREAD

**Issue by EdHallProper** [qtranslate/qtranslate-xt#1331]
Post pagination is broken for custom post type archives and causes critical errors when paging over.

Step by step:
1. navigate to a custom post type archive
2. click the next button on the post pagination
3. error appears

The Error message posted is as follows:  Uncaught Error: QTX_Module_Slugs::get_page_by_path(): Argument #1 ($page_path) must be of type string, null given, called in /www/wp-content/plugins/qtranslate-xt-master/src/modules/slugs/slugs.php on line 437
in /www/wp-content/plugins/qtranslate-xt-master/src/modules/slugs/slugs.php on line 1038

Version information:{
  "PHP_VERSION": "8.1.18",
  "WP_VERSION": "6.1.1",
  "QTX_VERSION": "3.15.0.dev.1",
  "Plugins": [
    "Query Monitor 3.11.1",
    "Advanced Custom Fields: OpenLayers Map 1.0.0",
    "Advanced Custom Fields PRO 6.1.4",
    "Classic Editor 1.6.3",
    "Cloudinary 3.0.7",
    "Crop Thumbnails 1.5.0",
    "Disable \"BIG image\" Threshold 1.0",
    "Event Organiser 3.12.3",
    "qTranslate-XT 3.14.2",
    "Relevanssi 4.19.0",
    "What The File 1.6.0",
    "WooCommerce Additional Variation Images 2.3.0",
    "WooCommerce Conditional Shipping and Payments 1.14.5",
    "WooCommerce Stripe Gateway 7.3.0",
    "WooCommerce Local Pickup Plus 2.9.12",
    "WooCommerce 7.6.0",
    "Yoast SEO 20.6",
    "WP REST Cache 2022.2.2",
    "WooCommerce PDF Invoices, Packing Slips, Delivery Notes &amp; Shipping Labels (Pro) 4.3.1",
    "Yoast Test Helper 1.17"
  ]
}

Configuration information: {
  "default_language": "en",
  "enabled_languages": [
    "en",
    "cy"
  ],
  "locale": {
    "en": "en_GB",
    "cy": "cy"
  },
  "date_format": {
    "en": "%A %B %e%q, %Y",
    "cy": "%e %B %G"
  },
  "time_format": {
    "en": "%I:%M %p",
    "cy": "%I:%M %p"
  },
  "url_mode": "4",
  "use_strftime": 3,
  "filter_options_mode": 0,
  "language_name_case": 0,
  "detect_browser_language": true,
  "hide_untranslated": false,
  "show_menu_alternative_language": false,
  "show_displayed_language_prefix": false,
  "show_alternative_content": false,
  "hide_default_language": false,
  "use_secure_cookie": false,
  "header_css_on": false,
  "text_field_filters": [],
  "domains": {
    "en": "garden.flywheelstaging.com",
    "cy": "garddfotaneg.cymru"
  },
  "disable_client_cookies": true,
  "url_info": {
    "cookie_lang_admin": "en",
    "cookie_front_or_admin_found": true,
    "scheme": "http",
    "host": "garden.flywheelstaging.com",
    "path": "/wp-admin/admin-ajax.php",
    "query": "action=admin_debug_info",
    "path-base": "",
    "wp-path": "/wp-admin/admin-ajax.php",
    "language_neutral_path": true,
    "http_referer": "http://garden.flywheelstaging.com/wp-admin/options-general.php?page=qtranslate-xt",
    "referer_admin": true,
    "doing_front_end": false,
    "lang_cookie_admin": "en",
    "lang_admin": "en",
    "language": "en",
    "set_cookie": false
  },
  "language": "en",
  "editor_mode": 0,
  "highlight_mode": 1,
  "auto_update_mo": true,
  "hide_lsb_copy_content": false,
  "lsb_style": "simple-tabs.css",
  "config_files": [
    "./i18n-config.json"
  ],
  "custom_fields": [],
  "custom_field_classes": [],
  "post_type_excluded": [
    "wc_pickup_location",
    "acf-field-group",
    "acf-field",
    "shop_order",
    "shop_order_refund",
    "shop_coupon"
  ],
  "admin_enabled_modules": {
    "acf": true,
    "woo-commerce": true,
    "wp-seo": true,
    "slugs": true,
    "all-in-one-seo-pack": false,
    "events-made-easy": false,
    "jetpack": false,
    "google-site-kit": false,
    "gravity-forms": false
  },
  "qtrans_compatibility": false
}

**Comment by yoancutillas** [comment]
This is fixing the issue in my case, thank you :)

**Comment by herrvigg** [comment]
Released in 3.15.2.
